### ISSUE TITLE
fix DeprecationWarning: IPv4Address.__getitem__ in protocols.ident

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4394]
|<img alt="forrestv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| forrestv reported|
|-|-|
|Trac ID|trac#4394|
|Type|defect|
|Created|2010-04-10 20:34:30Z|
|Branch|https://github.com/twisted/twisted/tree/ident-deprecation-4394|

IPv4Address.host should be used instead of IPv4Address[1] in IdentServer.validQuery

Attachments:

* [ident_DeprecationWarnings.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f1c/f1c6cfb5c5a959aee655e4f9250ca8303668074a/efc7f97504758018f62aed571c20ee5895ba7c83.diff) (776 bytes) - added by forrestv on 2010-04-10 20:34:50Z - patch to resolve issue

<details><summary>Searchable metadata</summary>

```
trac-id__4394 4394
type__defect defect
reporter__forrestv forrestv
priority__normal normal
milestone__ 
branch__branches_ident_deprecation_4394 branches/ident-deprecation-4394
branch_author__forrestv__exarkun forrestv, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ident ident
time__1270931670000000 1270931670000000
changetime__1275138154000000 1275138154000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra removed owner|
|-|-|

Putting it up for review.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|spiv set owner to spiv<br>spiv set status to `assigned`|
|-|-|

Seems fine to me.

It needs a news file (http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles) before it can be merged.  I think this would do:

4394.bugfix:
```
Fix DeprecationWarning: IPv4Address.__getitem__ in protocols.ident
```

I'll land this, although not tonight as I'm getting sleepy...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Hey, interesting.  When I first looked at this, I thought it needed tests because "trial twisted.test.test_ident" doesn't emit any warnings.  But 'trial --coverage' says the changed lines are covered after all.

Ah.  The unit tests use `StringTransport`, which has `getHost` and `getPeer` methods which return something that doesn't have this deprecation.  I filed [#4401](https://github.com/twisted/twisted/issues/4401) for that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

I'm not sure why no one noticed earlier, but the patch breaks two `test_ident` unit tests, precisely because of [#4401](https://github.com/twisted/twisted/issues/4401).  It might be good to resolve that ticket first, otherwise this requires more changes.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|spiv commented|
|-|-|

Thanks for noticing!  I'm happy to tackle [#4401](https://github.com/twisted/twisted/issues/4401), but I'm going to be very busy for the next few days so there's plenty of opportunity for someone else to do it first :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [29079]) Branching to 'ident-deprecation-4394'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [29080]) Apply ident_DeprecationWarnings.diff 

refs [#4394](https://github.com/twisted/twisted/issues/4394)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Stuck the patch in the branch, made a few more minor changes.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ident-deprecation-4394).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [29083]) Merge ident-deprecation-4394

Author: forrestv, exarkun
Reviewer: spiv, therve
Fixes: [#4394](https://github.com/twisted/twisted/issues/4394)

Change `twisted.protocols.ident` to use the `IPv4Address` object's
`host` attribute rather than using the deprecated indexing API.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
